addons/a11y/src/components/Report/HighlightToggle.tsx

```diff
@@ -136,42 +147,32 @@ class HighlightToggle extends Component<ToggleProps, {}> {
     this.props.addElement(payload);
   }
 
-  setTargetElementOutlineStyle(targetElement: HTMLElement, outlineStyle: string): void {
-    targetElement.style.outline = outlineStyle;
-  }
-
-  onToggle(): void {
-    for (let element of this.props.elementsToHighlight) {
-      if (element) {
-        const targetElement = getElementBySelectorPath(element.target[0]);
-        if (this.props.highlightedElementsMap.get(targetElement)) {
-          let originalOutline = this.props.highlightedElementsMap.get(targetElement).originalOutline;
-          if (
-            this.props.isToggledOn &&
-            this.props.highlightedElementsMap.get(targetElement).isHighlighted
-          ) {
-            this.higlightRuleLocation(targetElement, false);
-            this.saveElementDataToMap(targetElement, false, originalOutline, this.props.type);
-          } else if (
-            !this.props.isToggledOn &&
-            !this.props.highlightedElementsMap.get(targetElement).isHighlighted
-          ) {
-            this.higlightRuleLocation(targetElement, true);
-            this.saveElementDataToMap(targetElement, true, originalOutline, this.props.type);
-          }
-        }
+  onToggle = (): void => {
+    this.props.elementsToHighlight.forEach(element => {
+      const targetElement = getElementBySelectorPath(element.target[0]);
+      if (!this.props.highlightedElementsMap.has(targetElement)) {
+        return;
       }
-    }
-  }
+      const originalOutline = this.props.highlightedElementsMap.get(targetElement).originalOutline;
+      const { isHighlighted } = this.props.highlightedElementsMap.get(targetElement);
+      const { isToggledOn } = this.props;
+      if ((isToggledOn && isHighlighted) || (!isToggledOn && !isHighlighted)) {
+        const addHighlight = !isToggledOn && !isHighlighted;
+        this.highlightRuleLocation(targetElement, addHighlight);
+        this.saveElementDataToMap(targetElement, addHighlight, originalOutline);
+      }
+    });
+  };
 
   render() {
     return (
       <Checkbox
+        ref={this.checkBoxRef}
         id={this.props.toggleId}
         type="checkbox"
         aria-label="Highlight result"
-        disabled={this.props.elementsToHighlight && this.props.elementsToHighlight.length === 0 ? true : false}
-        onChange={() => this.onToggle()}
+        disabled={!this.props.elementsToHighlight.length}
```

Comment:
I think this will throw a null pointer if this.props.elementsToHighlight is null or undefined

Reply:
Yes. When you pass `elementsToHighlight` not array or not a valid type its error. Only array correct type for this prop. Given throw error. It will explicity.

Reply:
If it’s going to throw an error, why wouldn’t we check for the presence of the list

Reply:
If it’s going to throw an error, why wouldn’t we check for the presence of the list

Reply:
In which cases there will be an error? When nothing passed - fallback to `defaultProps`  value. When passed an incorrect value (string, number) - throw an error, because it's not correct way API use. And in type definitions I use strict def:   `elementsToHighlight: NodeResult[];`.

I think throwing error more explicitly than silent component working.

Reply:
I think you're right about trying to get some type of error if something goes wrong rather than it failing silently. Maybe that could be a Log instead but there are different opinions about how to accomplish this and Im not sure one is better than the other so Im going to approve.